File: VehicleStore/src/VehicleStore.Infrastructure/Data/Mappers/UserMapper.cs
using VehicleStore.Domain.Entities;
using VehicleStore.Domain.Enums;
using VehicleStore.Infrastructure.Data.Records;

namespace VehicleStore.Infrastructure.Data.Mappers;

internal static class UserMapper
{
    public static User ToDomain(UserRecord record) => new()
    {
        Id = record.Id,
        Username = record.Username,
        Email = record.Email,
        PasswordHash = record.PasswordHash,
        Role = Enum.Parse<UserRole>(record.Role, ignoreCase: true),
        CreatedAt = record.CreatedAt
    };

    public static UserRecord ToRecord(User user) => new()
    {
        Id = user.Id,
        Username = user.Username,
        Email = user.Email,
        PasswordHash = user.PasswordHash,
        Role = user.Role.ToString(),
        CreatedAt = user.CreatedAt
    };
}

File: vehicle-store-web/src/app/data/mappers/user.mapper.ts
import { UserResponse } from '../../domain/interfaces/responses/user.response';
import { User } from '../../domain/models/user.model';

export class UserMapper {
  static toDomain(response: UserResponse): User {
    return {
      id: response.id,
      username: response.username,
      email: response.email,
      role: response.role,
      createdAt: response.createdAt
    };
  }

  static toDomainList(responses: UserResponse[]): User[] {
    return responses.map(UserMapper.toDomain);
  }
}
